src/util/writer.cpp

```diff
@@ -248,6 +248,59 @@ Writer::write(const std::string& name, const sexp::Value& value)
   *out << ")\n";
 }
 
+void
+Writer::write_compressed(const std::string& name, const std::vector<unsigned int>& value, int width)
+{
+  indent();
+  *out << '(' << name;
+  if (width > 0)
+  {
+    *out << "\n";
+    indent();
+  }
+
+  int count = 0;
+  int multiplier = 0;
+  unsigned int multiplied_value = 0;
+  for (const auto& i : value)
+  {
+    const bool width_limit = (width > 0 && count >= width);
+
+    ++count;
+    if (multiplier > 0 && i == multiplied_value)
+    {
+      ++multiplier;
+    }
+    else
+    {
+      if (multiplier > 1)
+        *out << -multiplier << " " << multiplied_value << (width_limit ? "" : " ");
+      else if (multiplier == 1)
+        *out << multiplied_value << (width_limit ? "" : " ");
+
+      multiplier = 1;
+      multiplied_value = i;
+    }
+
+    if (width > 0 && count >= width)
+    {
+      if (multiplier > 1)
+        *out << -multiplier << " " << multiplied_value;
+      else if (multiplier == 1)
+        *out << multiplied_value;
+
+      count = 0;
+      multiplier = 0;
+      multiplied_value = 0;
+
+      *out << "\n";
```

Comment:
So, is there even a point in putting newlines anymore if we are now compressing? Seems pointless now.

Reply:
I think this should be discussed, if we don't want newlines and tilemap width separation in tiles, it can be removed.

Reply:
I mean, its just kinda unnecessary now... if we were using the spacing to actually align things.. well it doesnt matter if we are compressing it.